examples/differenceChart.html

```diff
@@ -0,0 +1,145 @@
+  var expandLegend = function() {
+      var exp = chart.multiChart.legend.expanded();
+      chart.multiChart.legend.align(false);
+      chart.update();
+  }
+</script>
+<svg id="chart2"></svg>
+<div style="height: 1000px; background-color: red;">
+
+</div>
+<script>
+  let actual, predicted, data, chart;
+
+  async function readTextFile(file) {
+    return new Promise((resolve) => {
+      var rawFile = new XMLHttpRequest();
+      rawFile.overrideMimeType('application/json');
+      rawFile.open('GET', file, true);
+      rawFile.onreadystatechange = function() {
+        if (rawFile.readyState === 4 && rawFile.status == '200') {
+          resolve(rawFile.responseText);
+        }
+      }
+      rawFile.send(null);
+    });
+  }
+
+  const keyForActualLessThanPredicted = 'Savings (kWh)';
+  const keyForActualGreaterThanPredicted = 'Excess Usage (kWh)';
+
+  async function loadData() {
+    actual = await readTextFile('./actual.json');
+    predicted = await readTextFile('./predicted.json');
+    const actualData = JSON.parse(actual);
+    const predictedData = JSON.parse(predicted);
+
+    data = [{
+      key: 'Actual usage (kWh)',
+      type: 'actual',
+      values: actualData[0].values,
+      color: 'black'
+    }, {
+      key: 'Predicted usage (kWh)',
+      type: 'expected',
+      values: predictedData.map(d => {
+        return {
+          recorded_at: d.falling_edge,
+          value: d.predicted
+        };
+      }),
+      color: 'blue'
+    }];
+  }
+
+  async function main() {
+    const colors = d3.scale.category20();
+
+    await loadData();
+
+    nv.addGraph(function() {
+      chart = nv.models.differenceChart();
+      chart
+        .keyForXValue('recorded_at')
+        .x(d => {
+          return new Date(d.recorded_at)
+        })
+        .y(d => d.value)
+        .keyForActualLessThanPredicted(keyForActualLessThanPredicted)
+        .keyForActualGreaterThanPredicted(keyForActualGreaterThanPredicted)
+        .height(500)
+        .showPredictedLine(true)
+        .interpolate('step-before')
+        .strokeWidth(0.5);
+
+      chart.multiChart.interactiveLayer.tooltip.contentGenerator(function (d) {
+        var header = d.value;
+        var headerhtml = "<thead><tr><td colspan='3'><strong class='x-value'>" + header + "</strong></td></tr></thead>";
+        var bodyhtml = "<tbody>";
+        var series = d.series;
+        series.forEach(function(c) {
+          var value = (c.value || 0).toFixed(2);
+          if (c.key === keyForActualGreaterThanPredicted || c.key === keyForActualLessThanPredicted) {
+            var diff = Math.abs(c.data.y0 - c.data.y1);
+            if (diff === 0) {
+              value = '-';
+            } else {
+              value = diff.toFixed(2);
+            }
+          }
+          bodyhtml = bodyhtml + "<tr><td class='legend-color-guide'><div style='background-color: " + c.color + ";'></div></td><td class='key'>" + c.key + "</td><td class='value'>" + value + "</td></tr>";
```

Comment:
This looks like it allows for XSS, you need to html escape the values you put within that html.  You can probably just copy the logic from lodash's escape method since we can't depend on jquery or something like that.  

https://github.com/lodash/lodash/blob/9d11b48ce5758df247607dc837a98cbfe449784a/escape.js

Reply:
I know this is the example, but people will just copy the examples to start their own work